src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,351 @@
+                "The third-body efficiency (eps) must be entered for M as"
+                " 'eps: {A: 1, b: 0, Ea: 0}' in reaction '{}'.", eqn);
+        }
+    }
+    if (colliders[0].hasKey("rate-constants")) {
+        rateObj_M = PlogRate(colliders[0], rate_units);
+        dataObj_M = PlogData();
+    }
+    else if (colliders[0].hasKey("Troe")) {
+        // Value of "type" is unimportant; just needed to make falloff.cpp run
+        colliders[0]["type"];
+        rateObj_M = TroeRate(colliders[0], rate_units);
+        dataObj_M = FalloffData();
+    }
+    else if (colliders[0].hasKey("pressure-range")) {
+        rateObj_M = ChebyshevRate(colliders[0], rate_units);
+        dataObj_M = ChebyshevData();
+    }
+    else {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Collider 'M' for reaction '{}' must be specified in a"
+            " pressure-dependent-Arrhenius (PLOG), falloff (Troe form), or Chebyshev"
+            " format.", eqn);
+    }
+    string eig_eps_key;
+    if (colliders[0].hasKey("eig0") && !colliders[0].hasKey("eps")) {
+        eig_eps_key = "eig0";
+    }
+    else if (colliders[0].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+        eig_eps_key = "eps";
+    }
+    else{
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Collider 'M' in reaction '{}' cannot contain both eps and eig0. Any"
+            " additional colliders must also make same choice as that of M.", eqn);
+    }
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
+    // Start at 1 because index 0 is for "M"
+    for (size_t i = 1; i < colliders.size(); i++){
+        if (!colliders[i].hasKey("name")) {
+                throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                    "The collider located at index '{}' in 'colliders' of reaction '{}'"
+                    " has no 'name' defined.", std::to_string(i), eqn);
+        }
+        auto nm = colliders[i]["name"].asString();
+        if (!colliders[i].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "Collider '{}' in reaction '{}' is missing an 'eps' or 'eig0' key.",
+                nm, eqn);
+        }
+        if (!colliders[i].hasKey(eig_eps_key)) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "All collision efficiencies in reaction '{}' must be defined uniformly"
+                " as either eig0 or eps. No mixing and matching is allowed.", eqn);
+        }
+        // Save data to colliderInfo, which will make it accessible by getParameters
+        colliderInfo.insert({colliders[i]["name"].as<string>(), colliders[i]});
+        colliderNames.push_back(colliders[i]["name"].as<string>());
```

Comment:
The simpler notation for inserting to a map is simply:
```suggestion
        colliderInfo[colliders[i]["name"].asString()] = colliders[i];
        colliderNames.push_back(colliders[i]["name"].as<string>());
```